website/content/docs/v0.9/Guides/converting-control-plane.md

```diff
@@ -0,0 +1,247 @@
+waiting for static pods for "kube-apiserver" to be present in the API server state
+waiting for static pods for "kube-controller-manager" to be present in the API server state
+waiting for static pods for "kube-scheduler" to be present in the API server state
+deleting daemonset "kube-apiserver"
+waiting for static pods for "kube-apiserver" to be present in the API server state
+deleting daemonset "kube-controller-manager"
+waiting for static pods for "kube-controller-manager" to be present in the API server state
+deleting daemonset "kube-scheduler"
+waiting for static pods for "kube-scheduler" to be present in the API server state
+conversion process completed successfully
+```
+
+As soon as control plane static pods are rendered, kubelet starts control plane static pods.
+Pods for `kube-apiserver` are crashing initially, as port :6443 is already occupied by self-hosted `kube-apiserver`.
+Script continues by removing each self-hosted daemonset and verifying that static pods are ready and healthy.
+
+## Manual Conversion
+
+Check that Talos runs self-hosted control plane:
+
+```bash
+$ talosctl -n <CONTROL_PLANE_IP> get bs
+NODE         NAMESPACE   TYPE              ID              VERSION   SELF HOSTED
+172.20.0.2   runtime     BootstrapStatus   control-plane   2         true
+```
+
+Talos machine configuration should be updated to the 0.9 format, there are two new required machine configuration settings:
+
+* `.cluster.serviceAccount` is the service account PEM-encoded private key.
+* `.cluster.aggregatorCA` is the aggregator CA for `kube-apiserver` (certficiate and private key).
+
+Current service account can be fetched from the Kubernetes secrets:
+
+```bash
+$ kubectl -n kube-system get secrets kube-controller-manager -o jsonpath='{.data.service\-account\.key}'
+LS0tLS1CRUdJTiBSU0EgUFJJVkFURS...
+```
+
+All control plane nodes machine configuration should be patched with the service account key:
+
+```bash
+$ talosctl -n <CONTROL_PLANE_IP1>,<CONTROL_PLANE_IP2>,... patch mc --immediate -p '[{"op": "add", "path": "/cluster/serviceAccount", "value": {"key": "LS0tLS1CRUdJTiBSU0EgUFJJVkFURS..."}}]'
+patched mc at the node 172.20.0.2
+```
+
+Aggregator CA can be generated using OpenSSL or any other certificate generation tools: RSA or ECDSA certificate with CN `front-proxy` valid for 10 years.
+PEM-encoded CA certificate and key should be base64-encoded and patched into the machine config at path `/cluster/aggregatorCA`:
+
+```bash
+$ talosctl -n <CONTROL_PLANE_IP1>,<CONTROL_PLANE_IP2>,... patch mc --immediate -p '[{"op": "add", "path": "/cluster/aggregatorCA", "value": {"crt": "S0tLS1CRUdJTiBDRVJUSUZJQ...", "key": "LS0tLS1CRUdJTiBFQy..."}}]'
+patched mc at the node 172.20.0.2
+```
+
+At this point static pod definitions and bootstrap manifests should be rendered, please see "Automated Conversion" on how to verify generated objects.
+Machine configuration can be updated until generated static pod definitions and bootstrap manifests look good.
```

Comment:
"Feel free to continue to refine your machine configuration until the generated static pod definitions and bootstrap manifests look good."